lib/services/livesync/livesync-service.ts

```diff
@@ -122,25 +114,27 @@ class LiveSyncService implements ILiveSyncService {
 
 	private partialSync(syncWorkingDirectory: string, onChangedActions: ((event: string, filePath: string, dispatcher: IFutureDispatcher) => void )[]): void {
 		let that = this;
-
-		let gazeWatcher = gaze("**/*", { cwd: syncWorkingDirectory, follow: true }, function (err: any, watcher: any) {
-			this.on('all', (event: string, filePath: string) => {
-				fiberBootstrap.run(() => {
-					that.$dispatcher.dispatch(() => (() => {
-						try {
-							for (let i = 0; i < onChangedActions.length; i++) {
-								onChangedActions[i](event, filePath, that.$dispatcher);
-							}
-						} catch (err) {
-							that.$logger.info(`Unable to sync file ${filePath}. Error is:${err.message}`.red.bold);
-							that.$logger.info("Try saving it again or restart the livesync operation.");
+		let pattern = ["app", "package.json", "node_modules"];
+		let watcher = choki.watch(pattern, { ignoreInitial: true, cwd: syncWorkingDirectory }).on("all", (event: string, filePath: string) => {
+			fiberBootstrap.run(() => {
+				that.$dispatcher.dispatch(() => (() => {
+					try {
+						filePath = path.join(syncWorkingDirectory, filePath);
+						for (let i = 0; i < onChangedActions.length; i++) {
+							onChangedActions[i](event, filePath, that.$dispatcher);
 						}
-					}).future<void>()());
-				});
+					} catch (err) {
+						that.$logger.info(`Unable to sync file ${filePath}. Error is:${err.message}`.red.bold);
+						that.$logger.info("Try saving it again or restart the livesync operation.");
+					}
+				}).future<void>()());
 			});
 		});
 
-		this.$processService.attachToProcessExitSignals(this, () => gazeWatcher.close());
+		this.$processService.attachToProcessExitSignals(this, () => {
+			watcher.unwatch(pattern);
+			process.exit();
```

Comment:
I was not able to stop the process without calling process.exit. Maybe there is something specific in chokidar preventing the process from exiting. I didn't observed any side effects while testing this code. Watching for changes in all files is important in order to support predictable livesync. So I am open for suggestions.